element/index.js

```diff
@@ -65,3 +66,35 @@ export function renderToString( element ) {
 
 	return renderToStaticMarkup( element );
 }
+
+/**
+ * Concat two React children objects
+ *
+ * @param  {?Object} children1 First children value
+ * @param  {?Object} children2 Second children value
+ *
+ * @return {Array}             The concatenation
+ */
+export function concatChildren( children1, children2 ) {
```

Comment:
Huh, yeah, I can reproduce that. Seems to be an issue at the point of calling `cloneElement` on the string though. This variation seems to work okay for me:

```js
/**
 * Concatenate two or more React children objects
 *
 * @param  {...?Object} childrens Set of children to concatenate
 * @return {Array}                The concatenated value
 */
export function concatChildren( ...childrens ) {
	return childrens.reduce( ( memo, children, i ) => {
		Children.forEach( children, ( child, j ) => {
			if ( 'string' !== typeof child ) {
				child = cloneElement( child, {
					key: [ i, j ].join()
				} );
			}

			memo.push( child );
		} );

		return memo;
	}, [] );
}
```